mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/LinkBankAccountScreen.kt

```diff
@@ -0,0 +1,327 @@
+                        onSelectBank
+                    )
+                }
+            else
+                Box(
+                    modifier = Modifier.fillMaxWidth().padding(top = 15.dp),
+                    contentAlignment = Alignment.Center
+                ) {
+                    Text(
+                        text = "No bank found",
+                        style = MaterialTheme.typography.subtitle1
+                    )
+                }
+        }
+    }
+}
+
+
+@Composable
+fun BankGrid(banks: List<Bank>, onSelectBank: (Bank) -> Unit) {
+    Box(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(200.dp)
+            .background(Color.White)
+            .shadow(elevation = 2.dp)
+    ) {
+        LazyVerticalGrid(columns = GridCells.Fixed(3)) {
+            items(banks.size) { index ->
+                BankItem(bank = banks[index], onSelectBank)
+            }
+        }
+    }
+}
+
+@Composable
+fun BankItem(bank: Bank, onSelectBank: (Bank) -> Unit) {
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier
+            .padding(8.dp)
+            .height(80.dp)
+            .clickable {
+                onSelectBank(bank)
+            }
+    ) {
+        Image(
+            painter = painterResource(id = bank.image),
+            contentDescription = bank.name,
+            modifier = Modifier.weight(1f)
+        )
+        Text(
+            text = bank.name,
+            style = MaterialTheme.typography.body1.copy(fontSize = 12.sp),
+            modifier = Modifier
+
+                .padding(top = 4.dp)
+        )
+    }
+}
+
+@Composable
+fun BankList(otherBanks: List<Bank>, onSelectBank: (Bank) -> Unit) {
+
+    Box(
+        modifier = Modifier
+            .background(Color.White)
+    ) {
+        Column(
+        ) {
+            otherBanks.forEachIndexed { index, bank ->
+                OtherBankItem(bank = bank, onSelectBank)
+                Divider(modifier = Modifier.padding(horizontal = 16.dp))
+            }
+        }
+    }
+}
+
+@Composable
+fun OtherBankItem(bank: Bank, onSelectBank: (Bank) -> Unit) {
+    Row(
+        horizontalArrangement = Arrangement.Start,
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = Modifier
+            .padding(vertical = 8.dp, horizontal = 10.dp)
+            .fillMaxWidth()
+            .clickable {
+                onSelectBank(bank)
+            }
+    ) {
+        Image(
+            painter = painterResource(id = bank.image),
+            contentDescription = bank.name,
+            modifier = Modifier.size(30.dp)
+        )
+        Spacer(modifier = Modifier.width(16.dp))
+        Text(text = bank.name, style = TextStyle(fontSize = 14.sp))
+    }
+}
+
+
+@Preview(showSystemUi = true, device = "id:pixel_5")
+@Composable
+private fun LinkBankAccountScreenPreview() {
+    MifosTheme {
+        LinkBankAccountScreen(onSelectBank = {
+
+        })
```

Comment:
change it to LinkBankAccountScreen(onSelectBank = })  no need to add so much of extra space